pyang/statements.py

```diff
@@ -1879,6 +1879,14 @@ def v_reference_list(ctx, stmt):
     stmt.i_is_validated = True
 
     def v_key():
+        def iterate_action(keyword, s):
```

Comment:
This is a bit inefficient (it will check nested lists more than once), and it doesn't solve the problem w/ actions in groupings or augmented actions, as you noted.
A better solution might be to add a new v_ancestor_list_key or something after expansion (probably a new validation_phase after unique_names is needed), where you follow each action's parent pointer to check for list keys.

Reply:
OK, I will try to handle this.